x/dex/contract/market_order.go

```diff
@@ -41,10 +41,15 @@ func getUnfulfilledPlacedMarketOrderIds(
 			continue
 		}
 		if order.OrderType == types.OrderType_MARKET || order.OrderType == types.OrderType_LIQUIDATION ||
-			order.OrderType == types.OrderType_FOKMARKET || order.OrderType == types.OrderType_FOKMARKETBYVALUE {
+			order.OrderType == types.OrderType_FOKMARKET {
 			if settledQuantity, ok := orderIDToSettledQuantities[order.Id]; !ok || settledQuantity.LT(order.Quantity) {
 				res = append(res, order.Id)
 			}
+		} else if order.OrderType == types.OrderType_FOKMARKETBYVALUE {
```

Comment:
nit: maybe worth to add a test case where the `FOKMARKETBYVALUE` order has a non-zero settlement quantity but the settled quantity is < order quantity

Reply:
sg, will create a separate ticket to add the test for FOKMARKETBYVALUE and FOKMARKET after the dex module_test.go is updated